addons/purchase/tests/test_purchase_invoice.py

```diff
@@ -467,3 +467,38 @@ def test_sequence_autocomplete_invoice(self):
         ]
         for line in invoice.invoice_line_ids.sorted('sequence'):
             self.assertEqual(line.purchase_order_id, expected_purchase.pop(0))
+
+    def test_partial_billing_interaction_with_invoicing_switch_threshold(self):
+        """ Let's say you create a partial bill 'B' for a given PO. Now if you change the
+            'Invoicing Switch Threshold' such that the bill date of 'B' is before the new threshold,
+            the PO should still take bill 'B' into account.
+        """
+        purchase_order = self.env['purchase.order'].with_context(tracking_disable=True).create({
+            'partner_id': self.partner_a.id,
+        })
+        line = self.env['purchase.order.line'].create({
+            'name': self.product_deliver.name,
+            'product_id': self.product_deliver.id,
+            'product_qty': 20.0,
+            'product_uom': self.product_deliver.uom_id.id,
+            'price_unit': self.product_deliver.list_price,
+            'order_id': purchase_order.id,
+            'taxes_id': False,
+        })
+
+        purchase_order.button_confirm()
+        line.qty_received = 10
+        action = purchase_order.action_create_invoice()
+
+        invoice = self.env['account.move'].browse(action['res_id'])
+        invoice.invoice_date = invoice.date
+        invoice.action_post()
+
+        self.assertEqual(line.qty_invoiced, 10)
+
+        invoice.state = 'cancel'
+        invoice.payment_state = 'invoicing_legacy'
```

Comment:
Thank you very much for the input! I only started about 5 weeks ago, so feedback is always welcome and appreciated. 

To be able to manipulate the invoicing threshold, I'll need the `account_accountant` module. So, I'll move the sale tests to the `sale_account_accountant` module, and I'll create a `purchase_account_accountant` module for the purchase tests.

This also means that those tests will be moved to a new enterprise PR. 
I'll request a review from you when this new PR is created.

Thanks again.

Reply:
Don't create a module only for one simple test :wink: 
Maybe just keep those tests where they are, but use https://docs.python.org/3/library/unittest.html#unittest.TestCase.skipTest if `account_accountant` is not installed.
A bit dirty, but I think it's a good tradeoff (better than a new module)

Reply:
Yeah, people in my team also suggested the same thing 😅.
Thanks!

Reply:
> Note that your version is triggering the compute because of the dependence on `invoice_lines.move_id.state`, so changing the `qty_received` afterwards is useless in this case

Updating `qty_received` is still necessary because `invoice_lines.move_id.state` is changed through a raw SQL query.